pkg/networkmanager/interfaces.js

```diff
@@ -1464,6 +1466,25 @@ function add_usage_monitor(iface) {
                            ];
 }
 
+function complete_settings(settings, device) {
+    if (!device) {
+        console.warn("No device to complete settings", JSON.stringify(settings));
+        return;
+    }
+
+    settings.connection.id = device.Interface;
+    settings.connection.uuid = generate_uuid();
+
+    if (device.DeviceType == 'ethernet') {
+        settings.connection.type = '802-3-ethernet';
+        settings.ethernet = { };
```

Comment:
This looks odd and at least warrants a comment - a function with the purpose to add missing properties shouldn't *delete* all existing settings?